packages/techdocs-common/src/helpers.test.ts

```diff
@@ -109,10 +126,230 @@ describe('parseReferenceAnnotation', () => {
   });
 });
 
+describe('getDirLocation', () => {
+  it.each`
+    techdocsRef                    | responseTarget
+    ${undefined}                   | ${undefined}
+    ${16}                          | ${undefined}
+    ${'.'}                         | ${'.'}
+    ${'dir:.'}                     | ${'.'}
+    ${'./relative'}                | ${'./relative'}
+    ${'dir:./relative'}            | ${'./relative'}
+    ${'dir:https://github.com...'} | ${'https://github.com...'}
```

Comment:
Is this intended?  It doesn't seem like a valid use-case.

Reply:
Yeah more or less. It _is_ possible to use it as long as your entity is registered by a `url`-type location, but I wouldn't recommended it. So we better remove it from the test. We could also add some special logic to recognize that it is a url and throw an error like in the `backstage.io/techdocs-ref: https://...` case, but I am not really keen of doing that since we can't reuse any logic as we did in the other case. After all, this value will probably lead to a rejection in a later stage. What do you think?

Reply:
Indeed, @dhenneke!  Yeah, perhaps it's fine to leave as-is here, but it would be great to handle this rejection in TechDocs backend.

Trying this value locally results in an anonymous `Error` in the UI and nothing in the build logs but `waiting for logs`.  Catching and forwarding the error when the `DocsBuilder` is instantiated in the `DocsSynchronizer` would close the loop, I think.

Reply:
> Catching and forwarding the error when the `DocsBuilder` is instantiated in the `DocsSynchronizer` would close the loop, I think.

Correct. I didn't thought about the possibility that the constructor can throw in #6341. When I move it into the `try {} catch {}` block I get the following result:

![image](https://user-images.githubusercontent.com/720821/126475910-24581dc9-82b8-49f3-92c1-192113bcf1cd.png)

Not the fanciest error, but the user has at least a chance to deduce what was going on.